app/controllers/campsites_controller.rb

```diff
@@ -3,16 +3,19 @@ class CampsitesController < ApplicationController
 
   def index
     authorize Campsite
-    if params[:query] && params[:query]["campsite-address"] != ""
+    if params_exist?
       query_params = params[:query]
-      address = query_params["address"]
-      @campsites = policy_scope(Campsite)
-      @campsites = Campsite.near(address, 100)
+      dates = query_params["date-field"].split(" to ")
+      start_date = DateTime.parse(dates[0])
+      end_date = DateTime.parse(dates[1])
+      search_dates = (start_date..end_date).to_a
+      address = query_params["campsite-address"]
+      @campsites = policy_scope(Campsite).near(address, 100)
```

Comment:
What if there are none in the search ?what do we return to avoid disclosing a world map